profiling/perfetto-connector/libperfetto-connector.cpp

```diff
@@ -0,0 +1,182 @@
+
+#include "perfetto.h"
+#include <unistd.h>
+#include <fcntl.h>
+#include <cstdio>
+#include <cstdlib>
+#include <fstream>
+#include <ios>
+#include <sstream>
+#include <string>
+#include <type_traits>
+#include <string_view>
+#include <vector>
+
+#define KOKKOSP_PUBLIC_API __attribute__((visibility("default")))
+
+extern "C" {
+void kokkosp_init_library(const int loadSeq, const uint64_t interfaceVer,
+                          const uint32_t, void*) KOKKOSP_PUBLIC_API;
+void kokkosp_finalize_library() KOKKOSP_PUBLIC_API;
+void kokkosp_begin_parallel_for(const char* name, const uint32_t,
+                                uint64_t*) KOKKOSP_PUBLIC_API;
+void kokkosp_end_parallel_for(const uint64_t) KOKKOSP_PUBLIC_API;
+void kokkosp_begin_parallel_scan(const char* name, const uint32_t,
+                                 uint64_t*) KOKKOSP_PUBLIC_API;
+void kokkosp_end_parallel_scan(const uint64_t) KOKKOSP_PUBLIC_API;
+void kokkosp_begin_parallel_reduce(const char* name, const uint32_t,
+                                   uint64_t*) KOKKOSP_PUBLIC_API;
+void kokkosp_end_parallel_reduce(const uint64_t) KOKKOSP_PUBLIC_API;
+void kokkosp_push_profile_region(const char* regionName) KOKKOSP_PUBLIC_API;
+void kokkosp_pop_profile_region() KOKKOSP_PUBLIC_API;
+}
+
+PERFETTO_DEFINE_CATEGORIES(
+    perfetto::Category("kokkos.kernels").SetDescription("Kokkos Kernel Events"),
+    perfetto::Category("kokkos.regions")
+        .SetDescription("Kokkos Region Events"));
+PERFETTO_TRACK_EVENT_STATIC_STORAGE();
+
+namespace {
+auto tracing_session = std::unique_ptr<perfetto::TracingSession>{};
+
+template <typename Tp>
+auto get_env(std::string_view _env_name, Tp _default) {
+  const char* _env_val = std::getenv(_env_name.data());
```

Comment:
I think that is potentially a bug.  `std::getenv` expects null-terminated character string.
I would pass a `std::string` and use `c_str()`.  This is not performance critical.

Reply:
Good point